pymatgen/core/surface.py

```diff
@@ -437,6 +436,97 @@ def from_dict(cls, d):
             site_properties=s.site_properties, energy=d["energy"]
         )
 
+    def tag_surface_sites(self, decimal=5):
```

Comment:
Do we need to tag, or will returning a list of indices of the surface sites suffice?

Also, how do we know the surface site is on the top or bottom surface? I think there are instances where you want to know that